src/main/kotlin/nl/avisi/structurizr/site/generatr/StructurizrUtilities.kt

```diff
@@ -7,10 +7,10 @@ import com.structurizr.model.SoftwareSystem
 import com.structurizr.view.ViewSet
 
 val Model.includedSoftwareSystems: List<SoftwareSystem>
-    get() = softwareSystems.filter { it.includedSoftwareSystem }
-
-val SoftwareSystem.includedSoftwareSystem
-    get() = this.location != Location.External
+    get() = if (softwareSystems.any { it.group != null })
```

Comment:
This will not behave exactly the same as `this.location != Location.External`. Reason for this is that `location` can be `null` when no enterprise boundary is defined in the Structurizr model, which will result in a site where all systems are included. The suggested change will make it so that a model without any groups will result in an empty site.

Apart from that, my idea with this tool has always been to closely follow the Structurizr concepts. With this change, we build in the assumption that people use a top-level group to specify an enterprise boundary and want to exclude systems outside of the enterprise boundary from the generated site.

I don't think this is a safe assumption. Right now, both concepts exist, and the safe assumption is that they are both being used "in the field". People's reasons for using groups could be anything, including, but not limited to designating an enterprise boundary.

From the Structurizr point of view, removing the enterprise boundary is a breaking change and I think we should treat this in the same way, and not try to retrofit this concept into another concept that is clearly meant to be more generic and reusable.

I do think that being able to exclude certain systems from the sidebar menu is a desirable feature. The idiomatic way, I think, would be to implement this by using tags. We could introduce a new view property (something like `generatr.site.excludeTags`), which can optionally be used to specify one or more tags which should be excluded from the sidebar menu.

Reply:
Yes, agree, using tags instead of our own heuristics sounds good!